File: src/backend/UserService.Application/DTO/Balance/TransactionDto.cs
namespace UserService.Application.DTO.Balance;

public record TransactionDto
{
    public string TransactionId { get; init; }
    public decimal Amount { get; init; }
    public string OperationType { get; init; }
    public DateTime CreatedAt { get; init; }
    public string? Comment { get; init; }
    public string Status { get; init; }

    public TransactionDto()
    {
        OperationType = string.Empty;
        Status = string.Empty;
    }
}

File: src/frontend/client-angular/src/app/core/models/user-service/entities/transaction.dto.ts
export enum OperationType {
    DEPOSIT = 'Deposit',
    WITHDRAW = 'Withdraw',
}

export enum TransactionStatus {
    PENDING = 'Pending',
    COMPLETED = 'Completed',
    FAILED = 'Failed',
}

export interface TransactionDto {
    transactionId: string;
    amount: number;
    operationType: OperationType;
    createdAt: Date;
    comment?: string;
    status: TransactionStatus;
}